src/Entity/Competition/Competition.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace BalticRobo\Api\Entity\Competition;
+
+use BalticRobo\Api\Model\Cms\AddCompetitionDTO;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity(repositoryClass="BalticRobo\Api\Repository\Competition\CompetitionRepository")
+ */
+class Competition
+{
+    /**
+     * @ORM\Id()
+     * @ORM\GeneratedValue()
+     * @ORM\Column(type="integer")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(type="string", length=64)
+     */
+    private $display_name;
+
+    /**
+     * @ORM\Column(type="string", length=64)
+     */
+    private $name;
+
+    /**
+     * @ORM\Column(type="smallint")
+     */
+    private $year;
+
+    /**
+     * @ORM\Column(type="smallint")
+     */
+    private $registrationType;
+
+    public static function createFromAddDTO(AddCompetitionDTO $dto): self
+    {
+        $entity = new self();
+        $entity->display_name = $dto->getDisplayName();
+        $entity->name = $dto->getName();
+        $entity->year = $dto->getYear();
+        $entity->registrationType = $dto->getRegistrationType();
+
+        return $entity;
+    }
+
+    public function getId(): ?int
```

Comment:
Don't allow to get `null` from Entities! Never!

https://ocramius.github.io/doctrine-best-practices/#/50 to 52